ts/components/conversation/ConversationHeader.tsx

```diff
@@ -313,4 +248,93 @@ export class ConversationHeader extends React.Component<Props> {
       </div>
     );
   }
+
+  private renderPublicMenuItems() {
+    const {
+      i18n,
+      isBlocked,
+      isMe,
+      isGroup,
+      isArchived,
+      isPublic,
+      onResetSession,
+      onSetDisappearingMessages,
+      // onShowAllMedia,
+      onShowGroupMembers,
+      onShowSafetyNumber,
+      onArchive,
+      onMoveToInbox,
+      timerOptions,
+      onBlockUser,
+      onUnblockUser,
+      hasNickname,
+      onClearNickname,
+      onChangeNickname,
+    } = this.props;
+
+    const disappearingTitle = i18n('disappearingMessages') as any;
+
+    const blockTitle = isBlocked ? i18n('unblockUser') : i18n('blockUser');
+    const blockHandler = isBlocked ? onUnblockUser : onBlockUser;
+
+    const disappearingMessagesMenuItem = (
+      <SubMenu title={disappearingTitle}>
+        {(timerOptions || []).map(item => (
+          <MenuItem
+            key={item.value}
+            onClick={() => {
+              onSetDisappearingMessages(item.value);
+            }}
+          >
+            {item.name}
+          </MenuItem>
+        ))}
+      </SubMenu>
+    );
+    const showMembersMenuItem = isGroup && (
+      <MenuItem onClick={onShowGroupMembers}>{i18n('showMembers')}</MenuItem>
+    );
+    const showSafetyNumberMenuItem = !isGroup &&
+      !isMe && (
+        <MenuItem onClick={onShowSafetyNumber}>
+          {i18n('showSafetyNumber')}
+        </MenuItem>
+      );
+    const resetSessionMenuItem = !isGroup && (
+      <MenuItem onClick={onResetSession}>{i18n('resetSession')}</MenuItem>
+    );
+    const blockHandlerMenuItem = !isMe && (
+      <MenuItem onClick={blockHandler}>{blockTitle}</MenuItem>
+    );
+    const changeNicknameMenuItem = !isMe && (
+      <MenuItem onClick={onChangeNickname}>{i18n('changeNickname')}</MenuItem>
+    );
+    const clearNicknameMenuItem = !isMe &&
+      hasNickname && (
+        <MenuItem onClick={onClearNickname}>{i18n('clearNickname')}</MenuItem>
+      );
+    const archiveConversationMenuItem = isArchived ? (
+      <MenuItem onClick={onMoveToInbox}>
+        {i18n('moveConversationToInbox')}
+      </MenuItem>
+    ) : (
+      <MenuItem onClick={onArchive}>{i18n('archiveConversation')}</MenuItem>
+    );
+
+    return (
+      !isPublic && (
```

Comment:
You can comletely remove this and add this check to the very top of the file to return early